hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -227,15 +246,22 @@ def more(self) -> bool:
         self.data.start_example(ONE_FROM_MANY_LABEL)
 
         if self.min_size == self.max_size:
+            # if we have to hit an exact size, draw unconditionally until that
+            # point, and no further.
             should_continue = self.count < self.min_size
         else:
             forced_result = None
             if self.force_stop:
+                # if our size is forced, we can't reject in a way that would
+                # cause us to differ from the forced size.
+                assert self.forced_size is None or self.count == self.forced_size
```

Comment:
`many` is an unusual application of `forced`, because intermediate draws may happen that we have no control over. I think the best we can do here is rely on the consumer to force the correct draws to `more()`, and not `reject()` in non-forced ways.